builtin/providers/aws/resource_aws_config_config_rule.go

```diff
@@ -0,0 +1,314 @@
+								ValidateFunc: validateMaxLength(256),
+							},
+							Set: schema.HashString,
+						},
+						"tag_key": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateMaxLength(128),
+						},
+						"tag_value": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateMaxLength(256),
+						},
+					},
+				},
+			},
+			"source": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"owner": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateConfigRuleSourceOwner,
+						},
+						"source_detail": {
+							Type:     schema.TypeSet,
+							Set:      configRuleSourceDetailsHash,
+							Optional: true,
+							MaxItems: 25,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"event_source": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"maximum_execution_frequency": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateConfigExecutionFrequency,
+									},
+									"message_type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"source_identifier": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateMaxLength(256),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsConfigConfigRulePut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).configconn
+
+	name := d.Get("name").(string)
+
+	sources := d.Get("source").([]interface{})
+	source := sources[0].(map[string]interface{})
+	sourceInput := configservice.Source{
+		Owner:            aws.String(source["owner"].(string)),
+		SourceIdentifier: aws.String(source["source_identifier"].(string)),
+	}
+	if details, ok := source["source_detail"]; ok {
+		sourceInput.SourceDetails = expandConfigRuleSourceDetails(details.(*schema.Set))
+	}
+
+	ruleInput := configservice.ConfigRule{
+		ConfigRuleName: aws.String(name),
+		Source:         &sourceInput,
+	}
+
+	scopes := d.Get("scope").([]interface{})
+	if len(scopes) > 0 {
+		ruleInput.Scope = expandConfigRuleScope(scopes[0].(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		ruleInput.Description = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("input_parameters"); ok {
+		ruleInput.InputParameters = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("maximum_execution_frequency"); ok {
+		ruleInput.MaximumExecutionFrequency = aws.String(v.(string))
+	}
+
+	input := configservice.PutConfigRuleInput{
+		ConfigRule: &ruleInput,
+	}
+	log.Printf("[DEBUG] Creating AWSConfig config rule: %s", input)
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
Always was during my test runs and experimenting, this is merely waiting for eventually consistent IAM, which (from my past experience) was never "inconsistent" for more than ~30sec.

Increasing the timeout means we'd also increase wait times for users that genuinely used incorrect IAM policies - i.e. you'd be waiting 5mins to realize that your policy is wrong or missing.